numpy/random/mtrand/randomkit.c

```diff
@@ -240,6 +246,101 @@ rk_random(rk_state *state)
     return y;
 }
 
+
+/*
+ * Returns an unsigned 64 bit random integer.
+ */
+NPY_INLINE static npy_uint64
+rk_uint64(rk_state *state)
+{
+    npy_uint64 upper = (npy_uint64)rk_random(state) << 32;
+    npy_uint64 lower = (npy_uint64)rk_random(state);
```

Comment:
I understand ;) The Mersenne twister would be in big trouble if it weren't true, as the output would depend on the size of long. It is hard to trace all the way through, but one must have faith. Sometimes. In any case, the current `rk_ulong` function makes the same assumption, although it uses a `|` instead of a `+`. Hmm, might be a trifle faster.